### DISCUSSION TITLE
Optimizing for SFTP download speeds

### DISCUSSION THREAD

**Discussion by LadderLogic** [sshnet/SSH.NET#1407] [Q&A]
My use case requires downloading ~60MB files using SftpClient. I've been tinkering with sftpClient buffer size, but unable to figure out the right optimization. I can download this file in 10 seconds via open ssh shell, but if I do sftpfilestream.ReadAsync it takes about a minute. How do I make the download process faster without fragmenting the large object heap? I have full control over the target linux box

**Comment by Rob-Hague** [comment]
SFTP download speeds are not great. The library has plenty of inefficiencies but perhaps there is a particular bottleneck or suboptimal methodology. It would take a bit of profiling to figure out. You could also compare with `SftpClient.DownloadFile`, which has a different methodology (for some reason)
